session.go

```diff
@@ -440,11 +440,13 @@ func (snd *sender) send(frame []byte) (open bool) {
 	open = snd.coalesceAdditionalFrames()
 
 	if snd.pingInterval > 0 {
+		snd.mx.Lock()
 		now := time.Now()
 		if now.Sub(snd.lastPing) > snd.pingInterval {
 			snd.bufferFrame(ping())
 			snd.lastPing = now
 		}
+		snd.mx.Unlock()
```

Comment:
Does this seem too aggressive?  We could use the read lock for the if statement.

Reply:
Maybe call `snd.bufferFrame` after unlock the mutex just in case if later changes cause `bufferFrame` to block or take longer?

Reply:
Also maybe have a separate mutex for `lastPing`?

Reply:
If performance is really a concern, we can turn `lastPing` into int64 (so we can use atomic operations) which is the result of `time.Now().UnixNano()`, and construct `time.Time` again by `time.Unix(0, snd.lastPing)`.

Reply:
All good ideas!  I didn't even think about atomic operations for this actually.